ui/src/message_popup/renderers/response_renderer_test.jsx

```diff
@@ -0,0 +1,70 @@
+/* flow weak */
+import React from 'react';
+
+import {shallow} from 'enzyme';
+import {expect} from 'chai';
+import sinon from 'sinon';
+import * as TestFixtures from '../test_fixtures.js';
+
+import ResponseRenderer from './response_renderer.jsx';
+import TextField from 'material-ui/TextField';
+import RaisedButton from 'material-ui/RaisedButton';
+import FeedbackCard from '../feedback_card.jsx';
+import Snackbar from 'material-ui/Snackbar';
+
+function testProps(props) {
+  return {
+    question: TestFixtures.testQuestion,
+    scaffolding: TestFixtures.practiceScaffolding,
+    limitMs: 30000,
+    onLogMessage: sinon.spy(),
+    onResponseSubmitted: sinon.spy(),
```

Comment:
A "spy" is an object that can be called like a function, but that just tracks when it was called and with what arguments.  It can also return fake values.  Here we're using it to be able to test whether these callbacks get called when typing in responses and hitting "save response."